source/mgnp-pitch/src/lib.rs

```diff
@@ -29,6 +47,48 @@ enum IdentityKind {
     Name(heapless::String<32>),
 }
 
-struct Registry {
+struct Registry {}
+
+impl<Fr, Wi, Lcl> Interface<Fr, Wi, Lcl>
+where
+    Fr: Frame,
+    Wi: Wire<Frame = Fr>,
+    Lcl: Wire<Frame = Fr>,
+{
+    pub async fn handle_connection(&mut self) -> Result<(), ()> {
+        let Self { wire, conn_table } = self;
+        loop {
+            let frame = futures::select_biased! {
+                // inbound frame from the wire.
+                frame = wire.recv().fuse() => frame?,
+                // either a connection needs to send data, or a connection has
+                // closed locally.
+                frame = conn_table.next_outbound().fuse() => {
+                    let frame: Fr = todo!("eliza: serialize outbound msg to frame; may need interface changes...");
```

Comment:
Not sure I totally understand, but a bidi tricky pipe should handle this.

The conn table will hold `Bidi<Frame, Frame>`, and the other side will hold:

* If it's an interface, it'll also hold `Bidi<Frame, Frame>`
* If it's a local service, it'll hold `Bidi<ToService, ToClient>`

Reply:
Specifically, it'll look kinda like this under the hood:

```
Bidi<Frame, Frame>    Bidi<TS, TC>
  rx <- (ser)   <- TC <-  tx
  tx -> (deser) -> TS ->  rx
```

The one nit is that at least in my current sketch, a conn drop means that the Bidi (or something holding it) needs to be be able to create a "close" Frame with the right ID.

Reply:
hmm, i think we still need to be doing a (tiny bit of) serializing either here or in the conn-table.
this is because, although each conn in the conn-table has a bidi of frames, in
my current implementation, the conn table will be responsible for selecting over
those bidis and determining their connection ID, so we will need to be writing
the ID here?

Reply:
that's sort of the tricky thing here that i have yet to figure out: if we're receiving a `&[u8]` frame from the bidi, how do we prefix that with a link ID without having to realloc? i think we may want to change the `Wire` trait so it's not a thing of `Frame`s but a thing you can write bytes to, so you can write the link ID and then the payload.